### PR TITLE
Make jax.numpy.squeeze as strict as numpy.squeeze about axis shape.

### PR DESCRIPTION
**Pull request by GeorgOstrovski**
Raise error if an axis explicitly selected to be squeezed has shape != 1. Related issue: #2284

### CODE REVIEW

**Comment by mattjj** [comment]
Beautiful, thanks for this!
